src/reducers/sources.js

```diff
@@ -464,19 +483,21 @@ export function hasPrettySource(state: OuterState, id: string) {
   return !!getPrettySource(state, id);
 }
 
+function getPlainUrl(url: string): string {
+  const queryStart = url.indexOf("?");
+  return queryStart !== -1 ? url.slice(0, queryStart) : url;
+}
+
 export function getSourcesUrlsInSources(
   state: OuterState,
   url: string
 ): string[] {
-  const urls = getUrls(state);
-  if (!url || !urls[url]) {
+  if (!url) {
     return [];
   }
-  const plainUrl = url.split("?")[0];
-
-  return Object.keys(urls)
-    .filter(Boolean)
-    .filter(sourceUrl => sourceUrl.split("?")[0] === plainUrl);
+  const plainUrl = getPlainUrl(url);
+  const plainUrls = getPlainUrls(state);
+  return plainUrls[plainUrl] || [];
```

Comment:
nit: i think this could be a selector

```js
function getPlainUrlForUrl(url) {
   const plainUrl = getPlainUrl(url);
   return getPlainUrls(state)[plainUrl] || [];
}
```

Reply:
Like so?
```javascript
function getPlainUrlSelectorForUrl(url) {
  const plainUrl = getPlainUrl(url);
  return createSelector(
    getPlainUrls,
    (plainUrls) => plainUrls[plainUrl] || [];
  );
}
```